File: Backend/src/main/java/com/example/backend/entity/RecipeContentEntity.java
package com.example.backend.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Entity
@Table(name = "recipe_contents")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class RecipeContentEntity {

    @Id
    private Integer id;

    @OneToOne
    @JoinColumn(name = "recipe_id", nullable = false)
    @MapsId
    private RecipeEntity recipe;

    @Column(name = "cooking_steps", columnDefinition = "JSONB")
    private String cookingSteps;

    @Column(length = 500)
    private String tips;
}

File: MyApplication/data/src/main/java/com/example/data/Recipes/model/RecipeContentEntity.kt
package com.example.data.Recipes.model

import com.example.domain.Recipes.model.ContentBlock
import com.example.domain.Recipes.model.RecipeContent
import com.example.domain.Recipes.model.GroceryItem

data class RecipeContentEntity(
    val recipeId: Int,
    val cookingSteps: List<ContentBlock>,
    val tips: String?
)